test/connectors/EnrolmentStoreConnectorSpec.scala

```diff
@@ -79,63 +78,63 @@ class EnrolmentStoreConnectorSpec extends SpecBase with MockitoSugar with ScalaF
               |}
               |]
               |}
-            """.stripMargin))))
+            """.stripMargin), headers = Map.empty[String, Seq[String]]))
         )
         result.futureValue mustBe Right(UserEnrolments(List(UserEnrolmentStatus("IR-PAYE", Some("active"), Some(new DateTime("2018-10-13T17:36:00.000").toLocalDateTime)),
           UserEnrolmentStatus("VAT", Some("active"), Some(new DateTime("2018-10-13T17:36:00.000").toLocalDateTime)),
           UserEnrolmentStatus("SA", Some("active"), Some(new DateTime("2018-10-13T17:36:00.000").toLocalDateTime)))))
       }
       "handle a 200 response with invalid JSON" in {
         when(httpGet.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any())).thenReturn(
-          Future.successful(HttpResponse(200, Some(Json.parse(
+          Future.successful(HttpResponse(status = 200, json = Json.parse(
             """
               |{
               |"enrolments":""
               |}
-            """.stripMargin))))
+            """.stripMargin), headers = Map.empty[String, Seq[String]]))
         )
         result.futureValue mustBe Left("Unable to parse data from enrolment API")
       }
       "handle a 404 response" in {
         when(httpGet.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any())).thenReturn(
-          Future.successful(HttpResponse(404, None))
+          Future.successful(HttpResponse.apply(404, None.toString))
         )
         result.futureValue mustBe Left("User not found from enrolment API")
       }
       "handle a 400 response" in {
         when(httpGet.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any())).thenReturn(
-          Future.successful(HttpResponse(400, None))
+          Future.successful(HttpResponse.apply(400, None.toString))
         )
         result.futureValue mustBe Left("Bad request to enrolment API")
       }
       "handle a 403 response" in {
         when(httpGet.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any())).thenReturn(
-          Future.successful(HttpResponse(403, None))
+          Future.successful(HttpResponse.apply(403, None.toString))
         )
         result.futureValue mustBe Left("Forbidden from enrolment API")
       }
       "handle a 503 response" in {
         when(httpGet.GET[HttpResponse](Matchers.any())(Matchers.any(), Matchers.any(), Matchers.any())).thenReturn(
-          Future.successful(HttpResponse(503, None))
+          Future.successful(HttpResponse.apply(503, None.toString))
         )
         result.futureValue mustBe Left("Unexpected error from enrolment API")
       }
```

Comment:
None of these work this way, so these tests don't actually give us confidence that this change hasn't broken something. They should be returning `Future.failed(UpstreamErrorResponse(...))` in order to correctly check the behaviour